sequencing/sequencer.go

```diff
@@ -656,7 +436,7 @@ func (c *Sequencer) GetNextBatch(ctx context.Context, req sequencing.GetNextBatc
 
 func (c *Sequencer) recover(batch sequencing.Batch, err error) (*sequencing.GetNextBatchResponse, error) {
 	// Revert the batch if Hash() errors out by adding it back to the BatchQueue
-	revertErr := c.bq.AddBatch(batch, c.db)
+	revertErr := c.bq.AddBatchToTheTop(batch, c.db)
```

Comment:
This is so we keep the ordering of the batches, although I'm not very sure that we need it at all. Why Hash() would fail? It's currently just marshalling and hashing:

```go
// Hash returns the hash of a batch.
func (batch *Batch) Hash() ([]byte, error) {
	batchBytes, err := batch.Marshal()
	if err != nil {
		return nil, err
	}
	hash := sha256.Sum256(batchBytes)
	return hash[:], nil
}
```